### PR TITLE
Commands by username, case insensitive usernames

### LINKED ISSUE #12: add user tag to /lasmatch and /recents
search last match and recents by user tag (i.e. /lastmatch @sampletelegramUser)

### PR DESCRIPTION
**Pull request by cschuijt**
This PR makes two changes:
- Allow users to ask for someone else's data when running /lastmatch, /recents or /rank. Simply add a telegram username (@ optional) to the command and the bot will try to match it to its database.
- Add case insensitivity to username storage. Usernames stored in the database from here on out will be in all-lowercase, and trying to find a user by their username will also convert to lowercase internally. ⚠ **This requires old capitalized usernames to be properly migrated!**

This PR also closes #12.

### CODE REVIEW

File: src/constants.py [hunk 1]
```diff
@@ -16,11 +16,12 @@
...
-    `\/recents <limit:optional>` :: Look up your most recent matches, defaults to 5 if limit is not defined\. Must have account id registered using `/register`\. Example :: `\/recents` or `\/recents 10` for 10 most recent matches\n
+    `\/recents <user:optional> <limit:optional>` :: Look up someone's most recent matches\. Defaults to 5 if limit is not defined and to you if user is not defined\. Must have account id registered using `/register`\. Example :: `\/recents`, `\/recents 10` for 10 most recent matches, `\/recents danvb` for Daniel's last games, `\/recents 20 KittyKirov` for Kirov's last 20 games\n
     `\/match <match_id>` :: Get detailed stats about the outcome of a match\n
-    `\/lastmatch` :: Gets the last match you played\. Must use register command prior to using this command
+    `\/lastmatch <user:optional>` :: Gets the last match someone played\. Defaults to you if no argument is given\. User must be registered for this to work \n
+    `\/rank <user:optional>` :: Gets a user's current medal\. Defaults to you if no argument is given\. User must be registered for this to work
```
Comment (connormullett) [inline src/constants.py:24]: Beautiful

File: src/bot/commands/user_commands.py [hunk 2]
```diff
@@ -33,26 +34,53 @@ def run_register_command(update, context):
 
 def run_get_player_recents_command(update, context):
     chat_id = update.message.chat_id
-    telegram_handle = update.message.from_user.username
 
-    registered_user = user_services.lookup_user_by_telegram_handle(telegram_handle)
```
Comment (connormullett) [inline src/bot/commands/user_commands.py:38]: these changes to commands that utilize `lookup_user_by_telegram_handle` are not converting it to lowercase. add it to the lookup service to make sure its consistent as all new users will be lowercase

File: src/bot/commands/user_commands.py [hunk 3]
```diff
@@ -33,26 +34,53 @@ def run_register_command(update, context):
...
+            update.message.from_user.username
+        )
+    else:
+        # Try to match the first argument against the database for usernames
+        # If the result is not a valid user, try the second one
+        # If the result still isn't valid, assume the user's own name
+        try:
+            registered_user = user_services.lookup_user_by_telegram_handle(
+                context.args[0].replace("@", "")
+            )
```
Comment (connormullett) [inline src/bot/commands/user_commands.py:50]: We can probably DRY this up by iterating over context.args

File: src/bot/commands/user_commands.py [hunk 4]
```diff
@@ -33,26 +34,53 @@ def run_register_command(update, context):
...
+    if not context.args:
+        registered_user = user_services.lookup_user_by_telegram_handle(
+            update.message.from_user.username
+        )
+    else:
+        # Try to match the first argument against the database for usernames
+        # If the result is not a valid user, try the second one
+        # If the result still isn't valid, assume the user's own name
+        try:
+            registered_user = user_services.lookup_user_by_telegram_handle(
```
Comment (connormullett) [inline src/bot/commands/user_commands.py:48]: lookup_user doesn't error when it fails to find a user. wrapping this in a try catch doesn't do anything. user lookup will return None if nothing is found

**Review by connormullett — CHANGES_REQUESTED** [review]
good job, couple potential oversights to change up and verify and should be good

File: src/bot/commands/user_commands.py (hunk 2 above)
Comment (cschuijt) [reply src/bot/commands/user_commands.py:38]: Lowercase conversion happens in `lookup_user_by_telegram_handle`, see user_services.py

File: src/bot/commands/user_commands.py (hunk 3 above)
Comment (cschuijt) [reply src/bot/commands/user_commands.py]: I've reworked /recents to iterate and update `registered_user`/`count` where appropriate

File: src/bot/commands/user_commands.py [hunk 5]
```diff
@@ -35,32 +35,20 @@ def run_register_command(update, context):
...
-        )
-    else:
-        # Try to match the first argument against the database for usernames
-        # If the result is not a valid user, try the second one
-        # If the result still isn't valid, assume the user's own name
+    # Assume defaults
+    registered_user = user_services.lookup_user_by_telegram_handle(
+        update.message.from_user.username
+    )
+    limit = constants.QUERY_PARAMETERS.RESPONSE_LIMIT.value
```
Comment (connormullett) [inline src/bot/commands/user_commands.py:42]: Nice

File: src/bot/commands/user_commands.py [hunk 6]
```diff
@@ -35,32 +35,20 @@ def run_register_command(update, context):
...
-            try:
-                registered_user = user_services.lookup_user_by_telegram_handle(
-                    context.args[1].replace("@", "")
-                )
-            except:
-                pass
-            if not registered_user:
-                registered_user = user_services.lookup_user_by_telegram_handle(
-                    update.message.from_user.username
-                )
```
Comment (connormullett) [inline src/bot/commands/user_commands.py]: So much cleaner

**Review by connormullett — APPROVED** [review]
Good job, thanks for the changes. Gimme 2 seconds to merge as I need to update prod

**Pull request merged**